frontend/libs/studio-components/src/components/StudioTextarea/StudioTextarea.test.tsx

```diff
@@ -0,0 +1,143 @@
+import type { StudioTextareaProps } from './StudioTextarea';
+import { StudioTextarea } from './StudioTextarea';
+import { act, render, screen } from '@testing-library/react';
+import type { RefObject } from 'react';
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+
+describe('StudioTextarea', () => {
+  it('Renders a textarea', () => {
+    renderTextarea();
+    expect(screen.getByRole('textbox')).toBeInTheDocument();
+  });
+
+  it('Renders with the given value', () => {
+    const value = 'test';
+    renderTextarea({ value });
+    expect(screen.getByRole('textbox')).toHaveValue(value);
+  });
+
+  it('Updates the value when the component rerenders with another value', () => {
+    const value = 'test';
+    const { rerender } = renderTextarea({ value });
+    expect(screen.getByRole('textbox')).toHaveValue(value);
+    const newValue = 'new value';
+    rerender(<StudioTextarea value={newValue} />);
+    expect(screen.getByRole('textbox')).toHaveValue(newValue);
+  });
+
+  it('Updates the value when the user types', async () => {
+    const user = userEvent.setup();
```

Comment:
That would reduce duplication and probably also increase performance. On the other hand, it's good practice in general to make the tests as isolated as possible, including instantiation of objects. But it wouldn't matter in this case indeed. I know that @WilliamThorenfeldt also have some opinions about this, what do you think?

Reply:
It is most convenient to include userEvent in every test. William and I discussed this earlier. Additionally, the RTL documentation is recommending this approach.

_"We recommend invoking userEvent.setup() before the component is rendered. This can be done in the test itself, or by using a setup function. We discourage rendering or using any userEvent functions outside of the test itself - e.g. in a before/after hook "_ Source: https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent